src/legacy/ui/public/saved_objects/components/saved_object_finder.tsx

```diff
@@ -52,31 +51,38 @@ interface SavedObjectFinderUIState {
   sortDirection?: Direction;
 }
 
-interface SavedObjectFinderUIProps extends InjectedIntlProps {
+interface BaseSavedObjectFinder {
   callToActionButton?: React.ReactNode;
   onChoose?: (
     id: SavedObject<SavedObjectAttributes>['id'],
     type: SavedObject<SavedObjectAttributes>['type']
   ) => void;
   makeUrl?: (id: SavedObject<SavedObjectAttributes>['id']) => void;
   noItemsMessage?: React.ReactNode;
-  savedObjectType: 'visualization' | 'search';
+  savedObjectType: 'visualization' | 'search' | 'index-pattern';
   visTypes?: VisTypesRegistryProvider;
+}
+
+interface SavedObjectFinderFixedPage extends BaseSavedObjectFinder {
+  initialPageSize?: undefined;
+  fixedPageSize: number;
+}
+
+interface SavedObjectFinderInitialPageSize extends BaseSavedObjectFinder {
   initialPageSize?: 5 | 10 | 15;
+  fixedPageSize?: undefined;
 }
+type SavedObjectFinderProps = SavedObjectFinderFixedPage | SavedObjectFinderInitialPageSize;
 
-class SavedObjectFinderUI extends React.Component<
-  SavedObjectFinderUIProps,
-  SavedObjectFinderUIState
-> {
+class SavedObjectFinder extends React.Component<SavedObjectFinderProps, SavedObjectFinderUIState> {
   public static propTypes = {
     callToActionButton: PropTypes.node,
     onChoose: PropTypes.func,
     makeUrl: PropTypes.func,
     noItemsMessage: PropTypes.node,
-    savedObjectType: PropTypes.oneOf(['visualization', 'search']).isRequired,
+    savedObjectType: PropTypes.oneOf(['visualization', 'search', 'index-pattern']).isRequired,
     visTypes: PropTypes.object,
-    initialPageSize: PropTypes.number,
+    pagination: PropTypes.object,
```

Comment:
I think this `pagination` prop type is now wrong. We could either simply use:

```ts
initialPageSize: PropTypes.oneOf([5, 10, 15]),
fixedPageSize: PropTypes.number,
```

Alternatively there is a more complex way of writing a custom prop validator that really checks if only one of it is set, but since we're switching over to TS in most places I don't think that's required for now.